### ISSUE TITLE
SmartAPI auto update not working correctly

### ISSUE THREAD

**Issue by kevinxin90** [SmartAPI/smartAPI#110]
The source SmartAPI spec for [MyGene](https://smart-api.info/editor/59dce17363dce279d389100834e43648) is updated 5 days ago on 2/24/2021. 

Based on the last-updated timestamp provided by [SmartAPI](https://smart-api.info/api/query?q=_id:59dce17363dce279d389100834e43648), it seems SmartAPI did spot there's a need to update on 2/24.

However, it doesn't seem to have applied the new updates from the source file. For examples, in the field `components/x-bte-response-mapping/functional_association_reactome`. It's currently a dict with two keys: **name, REACT**. However, in the SmartAPI registry, it's still NCBIGene.

So basically the need to update is successfully detected, however, the actual update doesn't seem to take place. Pls fix it.

**Comment by namespacestd0** [comment]
Server logged failure saving the new version in code 499. Confirmed the link at this moment indeed doesn't pass YAML validation with syntax error. Consider separating file corruption error from validation error to better indicate update status in the future. Keep this issue open to evaluate the effort involved.

**Comment by namespacestd0** [comment]
In summary, this is not a bug, the original document is malformed and the backend service correctly recognized it, saved its malformed status and refused to update the working document.
